Documentation/Design/Lambdas/Lambda Declare.md

```diff
@@ -0,0 +1,162 @@
+# Lambda Declare
+
+## Overview
+GLS will support declaring lambda functions. In order to declare a function that takes a lambda function as a parameter, the lambda must first be declared.
+
+## Commands
+
+### `lambda declare`
+`lambda declare : lambdaTypeName lambdaFunctionName returnType *`[parameterName parameterType ...]`*
+
+## Usage:
+
+```gls
+lambda declare : over21 check boolean age int
+```
+
+### CSharp:
+```CSharp
+delegate bool Over21(int age);
+```
+
+### Java:  
+```Java
+interface Over21 {
+    boolean check(int age);
+}
+```
+
+### Python:
+Python does not require declaring lambdas.
+
+### Ruby:
+Ruby does not require declaring lambdas.
+
+### TypeScript
+```TypeScript
+interface IOver21 {
+    (int age) : bool;
+}
+```
+
+## Implementation
+
+### Properties
+
+Properties will be stored in langauge.properties.lambdas. The following properties will be added.
+
+<table>
+    <thead>
+        <th>Property Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </thead>
+    <tbody>
+        <tr>
+            <td>requiresDeclaration</td>
+            <td>boolean</td>
+            <td>True if the language requires lambda declarations.</td>
```

Comment:
Here and below you've adopted the practice of `True if XYZ` / `True if XYZ, false otherwise`. I've always thought that type of phrasing was a little unnecessary: what are your thoughts of standardizing on `Whether XYZ`?